test/unit/schemaCompareWebViewController.test.ts

```diff
@@ -0,0 +1,1046 @@
+            database: undefined,
+            email: "sa@microsoft.com",
+            accountId: "",
+            tenantId: "",
+            port: 1433,
+            expiresOn: undefined,
+            encrypt: true,
+            hostNameInCertificate: undefined,
+            persistSecurityInfo: undefined,
+            columnEncryptionSetting: undefined,
+            attestationProtocol: undefined,
+            enclaveAttestationUrl: undefined,
+            commandTimeout: undefined,
+            connectRetryCount: undefined,
+            connectRetryInterval: undefined,
+            workstationId: undefined,
+            applicationIntent: undefined,
+            currentLanguage: "en-us",
+            pooling: undefined,
+            maxPoolSize: undefined,
+            minPoolSize: undefined,
+            loadBalanceTimeout: undefined,
+            replication: undefined,
+            attachDbFilename: undefined,
+            failoverPartner: undefined,
+            multiSubnetFailover: undefined,
+            multipleActiveResultSets: undefined,
+            packetSize: undefined,
+            typeSystemVersion: undefined,
+            connectionString: "",
+        };
+
+        treeNode = new TreeNodeInfo(
+            "localhost,1433, <default> (sa)",
+            context,
+            vscode.TreeItemCollapsibleState.None,
+            "localhost,1433",
+            null,
+            "Server",
+            "localhost,1433_NULL_sa_SqlLogin_trustServerCertificate:true_applicationName:vscode-mssql",
+            connInfo,
+            undefined,
+            null,
+        );
+
+        mockSchemaCompareService =
+            TypeMoq.Mock.ofType<mssql.ISchemaCompareService>();
+
+        vscodeWrapper = TypeMoq.Mock.ofType(
+            VscodeWrapper,
+            TypeMoq.MockBehavior.Loose,
+        );
+
+        mockConnectionManager = TypeMoq.Mock.ofType<ConnectionManager>();
+        mockConnectionManager
+            .setup((mgr) => mgr.getUriForConnection(TypeMoq.It.isAny()))
+            .returns(() => "localhost,1433_undefined_sa_undefined");
+
+        generateOperationIdStub = sandbox
+            .stub(scUtils, "generateOperationId")
+            .returns(operationId);
+
+        controller = new SchemaCompareWebViewController(
+            mockContext,
+            vscodeWrapper.object,
+            treeNode,
+            mockSchemaCompareService.object,
+            mockConnectionManager.object,
+            deploymentOptionsResult,
+            schemaCompareWebViewTitle,
+        );
+    });
+
+    teardown(() => {
+        generateOperationIdStub.restore();
+
+        sandbox.restore();
+    });
+
+    test("controller - initialize title - is 'Schema Compare'", () => {
+        assert.deepStrictEqual(
+            controller.panel.title,
+            schemaCompareWebViewTitle,
+            "Webview Title should match",
+        );
+    });
+
+    test("compare reducer - when called - runs once", async () => {
```

Comment:
Is there a reason to split apart the "runs once", "with correct arguments", and "returns expected results" tests?  I think those should usually be three assertions within the same test case.

Reply:
It's a personal preference of mine to break things apart like that, but I can shrink them down if you'd like.

Reply:
Also, I think it's easier to maintain these kinds of tests because when changes break the tests, determining what went wrong is practically immediate.